src/storage-client/src/storage_collections.rs

```diff
@@ -642,14 +649,16 @@ where
         // Construct the handle in a separate block to ensure all error paths
         // are diverging
         let since_handle = {
+            // If the collection we're openning is versioned, be sure to use a
+            // different CriticalId so the SinceHandles don't conflict.
+            let reader_id = match version {
```

Comment:
It's still not clear to me that we're managing the lifecycle of this handle properly... for example, I think the finalization task will only force-downgrade the handle of the controller-global handle, not these per-version handles. (And at a first pass it's not clear to me how all N critical handles get updated when the write frontier advances...)